aider/main.test.ts

```diff
@@ -0,0 +1,184 @@
+      agent_id: "foo",
+    });
+
+    // Install bash and run the script
+    const output = await executeScriptInContainerWithBash(state);
+
+    // Verify that the script at least attempted to set up Aider
+    expect(output.stdout).toContain("Setting up Aider AI pair programming...");
+  });
+
+  it("uses tmux when specified", async () => {
+    const state = await runTerraformApply(import.meta.dir, {
+      agent_id: "foo",
+      use_tmux: true,
+      use_screen: false,
+    });
+
+    // Instead of running the script, just verify the script content
+    // to ensure the tmux parameter is being properly applied
+    const instance = findResourceInstance(state, "coder_script");
+
+    // Check for the correct tmux condition with the interpolated value
+    expect(instance.script.includes('if [ "true" = "true" ]')).toBe(true);
+
+    // Also check for a unique string only present when tmux is used in the script
+    expect(instance.script.includes("tmux new-session -d -s")).toBe(true);
+  });
+
+  it("configures task reporting when enabled", async () => {
+    const state = await runTerraformApply(import.meta.dir, {
+      agent_id: "foo",
+      experiment_report_tasks: true,
+    });
+
+    // Install bash and run the script
+    const output = await executeScriptInContainerWithBash(state);
+
+    // Verify task reporting is mentioned
+    expect(output.stdout).toContain(
+      "Configuring Aider to report tasks via Coder MCP...",
+    );
+  });
+
+  it("passes aider cmd with correct flags when prompt is provided", async () => {
+    const state = await runTerraformApply(import.meta.dir, {
+      agent_id: "foo",
+    });
+
+    // Define a test prompt
+    const testPrompt = "Add a hello world function";
+
+    // Set up the environment variable for the task prompt
+    const output = await executeScriptInContainerWithBash(
+      state,
+      "alpine",
+      `echo 'export CODER_MCP_AIDER_TASK_PROMPT="${testPrompt}"' > /tmp/env_vars.sh`,
+    );
+
+    // Check if script contains the proper command construction with all required flags
+    const instance = findResourceInstance(state, "coder_script");
+
+    // Verify all required flags are present in the aider command
+    expect(
+      instance.script.includes(
+        "aider --architect --yes-always --read CONVENTIONS.md --message",
+      ),
+    ).toBe(true);
+
+    // Verify the expected message format is correct
+    expect(
+      instance.script.includes(
+        '--message \\"Report each step to Coder. Your task: $CODER_MCP_AIDER_TASK_PROMPT\\"',
+      ),
+    ).toBe(true);
+
+    // Verify the script creates a flag file to prevent duplicate execution
+    expect(instance.script.includes('touch "$HOME/.aider_task_executed"')).toBe(
+      true,
+    );
+
+    // Verify the app status slug is properly set in the screen session
+    expect(
+      instance.script.includes('export CODER_MCP_APP_STATUS_SLUG=\\"aider\\"'),
+    ).toBe(true);
```

Comment:
Why are we checking that the script contains these strings? Is it to prevent typos when it's edited in the future?